src/lib.rs

```diff
@@ -44,3 +44,13 @@ mod keyedvec;
 pub use self::codec::{Input, Output, Encode, Decode, Codec, Compact, HasCompact};
 pub use self::joiner::Joiner;
 pub use self::keyedvec::KeyedVec;
+
+/// Decode an object from the given slice.
+pub fn from_slice<R: Decode>(data: &[u8]) -> Option<R> {
+	R::decode(&mut &data[..])
+}
+
+/// Decode an object from the given vector.
+pub fn from_vec<R: Decode>(data: Vec<u8>) -> Option<R> {
```

Comment:
Do we actually need this one? Seems like superflous to me

Reply:
I think in certain situations this can be helpful. 
I wanted to `decode` something in a macro, without wanting to know the type it should decode to.
With this function I can just write `from_vec(vec)` and the compiler itself finds out the return type.
`Serde` for example provides similar functions.

Reply:
Such APIs can easily become a performance trap.

Can you please elaborate on the helpful situation a bit? Maybe an example.

Reply:
Why should that be a performance trap? I would never expect the compiler to not optimize it away?

Reply:
For example a user has a vector `v`, calls this without knowing about the `from_slice` thus cloning `v`. It is just not intuitive taking something non-Copy by value if you do not need it.